rest_framework/schemas.py

```diff
@@ -256,6 +256,289 @@ def get_allowed_methods(self, callback):
+            return []
+
+        fields = []
+        for field in serializer.fields.values():
+            if field.read_only or isinstance(field, serializers.HiddenField):
+                continue
+
+            required = field.required and method != 'PATCH'
+            field = coreapi.Field(
+                name=field.field_name,
+                location='form',
+                required=required,
+                schema=field_to_schema(field)
+            )
+            fields.append(field)
+
+        return fields
+
+    def get_pagination_fields(self, path, method):
+        view = self.view
+
+        if not is_list_view(path, method, view):
+            return []
+
+        pagination = getattr(view, 'pagination_class', None)
+        if not pagination:
+            return []
+
+        paginator = view.pagination_class()
+        return paginator.get_schema_fields(view)
+
+    def get_filter_fields(self, path, method):
+        view = self.view
+
+        if not is_list_view(path, method, view):
+            return []
+
+        if not getattr(view, 'filter_backends', None):
+            return []
+
+        fields = []
+        for filter_backend in view.filter_backends:
+            fields += filter_backend().get_schema_fields(view)
+        return fields
+
+    def get_encoding(self, path, method):
+        """
+        Return the 'encoding' parameter to use for a given endpoint.
+        """
+        view = self.view
+
+        # Core API supports the following request encodings over HTTP...
+        supported_media_types = set((
+            'application/json',
+            'application/x-www-form-urlencoded',
+            'multipart/form-data',
+        ))
+        parser_classes = getattr(view, 'parser_classes', [])
+        for parser_class in parser_classes:
+            media_type = getattr(parser_class, 'media_type', None)
+            if media_type in supported_media_types:
+                return media_type
+            # Raw binary uploads are supported with "application/octet-stream"
+            if media_type == '*/*':
+                return 'application/octet-stream'
+
+        return None
+
+# Note: With `AutoSchema` defined we attach it to APIView.
+#   * We do this here to avoid the dependency cycle from SchemaView needing
+#     APIView (below).
+#   * This requires importing _something_ from `rest_framework.schemas` or
+#     `rest_framework.documentation` before `APIView.schema will be available.
+#       * ???: When would `APIView.schema` be needed and that NOT be the case?
+#   * The alternative is to import AutoSchema to `views`, make `schemas` a
+#     package, and move SchemaView to `schema.views`, importing APIView there.
+APIView.schema = AutoSchema()
+
+
+class ManualSchema(ViewInspector):
+    """
+    Overrides get_link to return manually specified schema.
+    """
+    def __init__(self, link):
+        assert isinstance(link, coreapi.Link)
+        self._link = link
```

Comment:
Is `ManualSchema(link)` what we want, or is `ManualSchema({'GET': ..., 'POST': ...)` more likely to actually be useful? How about when applied to viewsets - is there any way we could target per-action? Not necessarily a blocker, but worth considering before we pull this in.

Reply:
Yeah. There's thought still to go into this one. 

Related issues: #5382 #4685 ...

My thought would be to address this Next™. Is this the right framework? If so pull it in and I'll turn the effort to the other Schema Generation tickets before 3.7